pkg/serverless/protocol.go

```diff
@@ -254,20 +246,22 @@ func (d *Daemon) EnableLogsCollection() (string, chan *logConfig.ChannelMessage,
 
 // StartInvocation tells the daemon the invocation began
 func (d *Daemon) StartInvocation() {
-	d.InvcWg.Add(1)
+	d.doneChannel = make(chan bool)
```

Comment:
You can specify a length of the channel to make it buffered and non blocking if you want.
https://tour.golang.org/concurrency/3

Reply:
@DarcyRaynerDD for my own curiosity -- what is the case for making this channel buffered and non-blocking?

Reply:
You wouldn't have to spin off `doneChannel <- true` into its own goroutine. An alternative would be to close `doneChannel` rather than sending on it. I think closes are always immediate.